rust/headless-client/src/lib.rs

```diff
@@ -73,14 +73,19 @@ struct Cli {
 impl Cli {
     fn command(&self) -> Cmd {
         // Needed for backwards compatibility with old Docker images
-        self.command.unwrap_or(Cmd::Standalone)
+        self.command.unwrap_or(Cmd::Auto)
     }
 }
 
 #[derive(clap::Subcommand, Clone, Copy)]
 enum Cmd {
+    /// If there is a token on disk, run in standalone mode. Otherwise, run as an IPC daemon
+    Auto,
```

Comment:
I don't like this behavior because of Hyrum's Law, I'd prefer to have `firezone-headless-client` bail out with a help message like `git` does.

But it was needed to share a systemd service unit between two tests that needs both subcommands. 

Depending how #4531 is settled, we might be able to remove this.